src/api/searchApi.ts

```diff
@@ -94,14 +82,60 @@ export const fetchNviCandidate = async (identifier: string) => {
   return getNviCandidates.data;
 };
 
-export const fetchResults2 = async (results: number, from: number, { doi = '' }) => {
-  const paginationQuery = `results=${results}&from=${from}`;
-  const doiQuery = doi ? `doi="${doi}"` : '';
+export interface FetchResultsQuery {
+  doi?: string;
+  identifier?: string;
+  contributor?: string;
+  query?: string;
+  category?: PublicationInstanceType | PublicationInstanceType[];
+  issn?: string;
+  publicationYear?: string;
+  title?: string;
+  project?: string;
+}
+
+export const fetchResults = async (
+  results: number,
+  from: number,
+  { doi, identifier, contributor, query, category, issn, publicationYear, title, project }: FetchResultsQuery
+) => {
+  let fullQuery = `results=${results}&from=${from}`;
 
-  const fullQuery = [paginationQuery, doiQuery].filter(Boolean).join('&');
+  if (title) {
+    fullQuery += `&title=${title}`;
+  }
+  if (doi) {
+    fullQuery += `&doi=${doi}`;
+  }
+  if (identifier) {
+    fullQuery += `&id=${identifier}`;
+  }
+  if (contributor) {
+    fullQuery += `&contributor=${contributor}`;
+  }
+  if (query) {
+    fullQuery += `&query=${query}`;
+  }
+  if (category) {
+    if (category instanceof Array) {
+      category.forEach((categoryValue) => (fullQuery += `&category=${categoryValue}`));
```

Comment:
Virker ikke atm.
Ser ut som bare den siste parameteren brukes når man bruker flere, søk på BookAnthology og AcademicArticle skal per nå returnere 3 treff, i stedet:
- Returnerer en BookAnthology: https://api.dev.nva.aws.unit.no/search/resources2?category=AcademicChapter&category=BookAnthology
- Returnerer to AcademicArticle: https://api.dev.nva.aws.unit.no/search/resources2?category=BookAnthology&category=AcademicChapter